src/vault.rs

```diff
@@ -135,6 +144,20 @@ impl Client {
         Ok(())
     }
 
+    pub fn kubernetes_auth(&mut self, role: String) -> Result<(), Error> {
```

Comment:
This doesn't need to take an owned copy of the `role` argument, it would work fine with just a reference, like `pub fn kubernetes_auth(&mut self, role:&str) -> Result<(), Error> {`. That will require a tweak to `AuthKubernetesRequest`, see the comment on `AuthKubernetesRequest`.